clients/drcachesim/tests/scheduler_unit_tests.cpp

```diff
@@ -2359,6 +2359,217 @@ test_replay_skip()
+        scheduler_t::scheduler_options_t sched_ops(scheduler_t::MAP_TO_ANY_OUTPUT,
+                                                   scheduler_t::DEPENDENCY_IGNORE,
+                                                   scheduler_t::SCHEDULER_DEFAULTS,
+                                                   /*verbosity=*/2);
+        zipfile_ostream_t outfile(record_fname);
+        sched_ops.schedule_record_ostream = &outfile;
+        if (scheduler.init(sched_inputs, NUM_OUTPUTS, sched_ops) !=
+            scheduler_t::STATUS_SUCCESS)
+            assert(false);
+        std::vector<std::thread> threads;
+        threads.reserve(NUM_OUTPUTS);
+        for (int i = 0; i < NUM_OUTPUTS; ++i) {
+            threads.emplace_back(std::thread(simulate_core, scheduler.get_stream(i),
+                                             &record_instr_count[i]));
+        }
+        for (std::thread &thread : threads)
+            thread.join();
+        if (scheduler.write_recorded_schedule() != scheduler_t::STATUS_SUCCESS)
+            assert(false);
+    }
+    {
+        // Replay.
+        std::vector<scheduler_t::input_workload_t> sched_inputs;
+        for (int i = 0; i < NUM_INPUTS; i++) {
+            std::vector<scheduler_t::input_reader_t> readers;
+            readers.emplace_back(
+                std::unique_ptr<mock_reader_t>(new mock_reader_t(inputs[i])),
+                std::unique_ptr<mock_reader_t>(new mock_reader_t()), BASE_TID + i);
+            sched_inputs.emplace_back(std::move(readers));
+            sched_inputs.back().thread_modifiers.push_back(
+                scheduler_t::input_thread_info_t(regions));
+        }
+        scheduler_t scheduler;
+        scheduler_t::scheduler_options_t sched_ops(scheduler_t::MAP_AS_PREVIOUSLY,
+                                                   scheduler_t::DEPENDENCY_IGNORE,
+                                                   scheduler_t::SCHEDULER_DEFAULTS,
+                                                   /*verbosity=*/2);
+        zipfile_istream_t infile(record_fname);
+        sched_ops.schedule_replay_istream = &infile;
+        if (scheduler.init(sched_inputs, NUM_OUTPUTS, sched_ops) !=
+            scheduler_t::STATUS_SUCCESS)
+            assert(false);
+        std::vector<uint64_t> replay_instr_count(NUM_OUTPUTS, 0);
+        ;
+        std::vector<std::thread> threads;
+        threads.reserve(NUM_OUTPUTS);
+        for (int i = 0; i < NUM_OUTPUTS; ++i) {
+            threads.emplace_back(std::thread(simulate_core, scheduler.get_stream(i),
+                                             &replay_instr_count[i]));
+        }
+        for (std::thread &thread : threads)
+            thread.join();
+        for (int i = 0; i < NUM_OUTPUTS; ++i) {
+            std::cerr << "Output #" << i << " recorded " << record_instr_count[i]
+                      << " instrs vs replay " << replay_instr_count[i] << " instrs\n";
+            assert(replay_instr_count[i] == record_instr_count[i]);
```

Comment:
This was reproducing an internal test which only checked the counts.  We could add a check of a schedule string.   The one I had added here for the tiny test case had the instr PC in there too which would be too hard to do when scaled up here.  It could look like the other tests with testing the input ordering. Done, added a string and a compare.